hpat/tests/test_series.py

```diff
@@ -2612,5 +2612,39 @@ def test_series_count_impl(S):
             result = hpat_func(S)
             self.assertEqual(result, result_ref)
 
+
+    def test_series_pct_change(self):
+        def test_series_pct_change_impl(S, periods, method):
+            return S.pct_change(periods, method)
+
+        hpat_func = hpat.jit(test_series_pct_change_impl)
+        test_input_data = [
+            [],
+            [np.nan, np.nan, np.nan],
+            [np.nan, np.nan, np.inf],
+            [0] * 8,
+            [0, 0, 0, np.nan, np.nan, 0, 0, np.nan, np.inf, 0, 0, np.inf, np.inf],
+            [1.1, 0.3, np.nan, 1, np.inf, 0, 1.1, np.nan, 2.2, np.inf, 2, 2]
+        ]
+        for input_data in test_input_data:
+            S = pd.Series(input_data)
+            for periods in [0, 1, 2, 5, 10, -1, -2, -5]:
+                for method in [None, "pad", "ffill", "backfill", "bfill"]:
+                    print(periods)
+                    print(input_data)
+                    result_ref = test_series_pct_change_impl(S, periods, method)
+                    result = hpat_func(S, periods, method)
+                    np.testing.assert_almost_equal(result, result_ref)
+
+    def test_series_pct_change2(self):
+        def test_series_pct_change_impl(S):
+            return S.pct_change()
+
+        hpat_func = hpat.jit(test_series_pct_change_impl)
+        S = pd.Series(["abababa", 6, "aaaaa", np.nan])
```

Comment:
I don't think it's correctly to mix data of different types inside of Series. Moreover you could use globally defined data, e.g.: https://github.com/IntelPython/sdc/commit/f2435b037ecc403c6adada060b94279c6ac2ca31#diff-deca39d332649cea819383154a5d2cb3R55-R62.

Reply:
> Moreover you could use globally defined data, e.g.: [f2435b0#diff-deca39d332649cea819383154a5d2cb3R55-R62](https://github.com/IntelPython/sdc/commit/f2435b037ecc403c6adada060b94279c6ac2ca31#diff-deca39d332649cea819383154a5d2cb3R55-R62).

Moreover you could use globally defined data, e.g.: f2435b0#diff-deca39d332649cea819383154a5d2cb3R55-R62.